stumpy/motifs.py

```diff
@@ -365,6 +366,11 @@ def match(
         The absolute tolerance parameter. This value will be added to `max_distance`
         when comparing distances between subsequences.
 
+    query_idx : int, default None
+        This parameter is used when the matrix profile is a self-join.
```

Comment:
This is great! I might consider elaborating or providing a little bit more context like:

```
This is the index position along the time series, `T`, where the query subsequence, `Q`, is located. `query_idx` should only be used when the matrix profile is a self-join and should be set to `None` for matrix profiles computed from AB-joins. If `query_idx` is set to a specific integer value, then this will help ensure that the self-match will be returned first.
```